app/services/fastly_vcl/whitelisted_params.rb

```diff
@@ -0,0 +1,65 @@
+module FastlyVCL
+  # Handles updates to our VCL snippet on Fastly that whitelists params
+  class WhitelistedParams
+    VCL_DELIMITER_START = "^(".freeze
+    VCL_DELIMITER_END = ")$".freeze
+    SNIPPET_NAME = "Whitelist certain querystring parameters".freeze
+    FILE_PARAMS = YAML.load_file("config/fastly/whitelisted_params.yml").sort.freeze
+
+    class << self
+      def update
+        fastly = Fastly.new(api_key: ApplicationConfig["FASTLY_API_KEY"])
+        service = fastly.get_service(ApplicationConfig["FASTLY_SERVICE_ID"])
+        latest_version = service.version
+
+        snippet = fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"],
+                                     latest_version.number,
+                                     SNIPPET_NAME)
+
+        current_params = params_to_sorted_array(snippet.content)
+
+        return if current_params == FILE_PARAMS # No update needed
+
+        new_version = latest_version.clone
+        new_snippet = fastly.get_snippet(ApplicationConfig["FASTLY_SERVICE_ID"],
+                                         new_version.number,
+                                         SNIPPET_NAME)
+        new_snippet.content = build_content(FILE_PARAMS, new_snippet.content)
+        new_snippet.save!
+
+        new_version.activate!
+        Rails.logger.info("Fastly updated to version #{new_version.number}.")
+        log_params_diff_to_datadaog(current_params, new_version)
+
+        true
```

Comment:
`log_params_diff_to_datadog` returns `1` if it's successful so I figured it might be better to just explicitly return `true` from this task.

Reply:
Is there aby way in which it could be not successful, i.e. would it maybe make sense to do the following instead of returning `true` to avoid accidentally masking a failure?

```ruby
log_params_diff_to_datadaog(current_params, new_version) == '1'
```

Reply:
Hmmm, I'll think about this one. This checks if the logging is successful which is more a nice to have than a deal-breaker. If the important operations failed, they raise a `Fastly::Error` before we get to this. I don't know that I'd want this whole task to return false if Datadog goofed somehow.